File: backend/dataset/maxPathSum/maxPathSum_15.py
class Solution:
    def maxPathSum(self, root: TreeNode) -> int:
        if root.right==None and root.left==None:
            return root.val
        ans=[]
        def recur(node):
            if node==None:
                return 0
            left_max=0
            right_max=0
            left_max+=recur(node.left)
            right_max+=recur(node.right)
            ans.append(max(node.val+right_max,left_max+node.val,node.val+right_max+left_max,node.val))
            return max(left_max+node.val,right_max+node.val,node.val)
        recur(root)
        print(ans)
        return max(ans)

File: backend/CDataset/maxPathSum/maxPathSum_15.cpp
class Solution {
public:
    #include<vector>
    class TreeNode{
        public:
        TreeNode* left;
        TreeNode* right;
        int val;
    };
    int maxPathSum(TreeNode* root) {
        int max_sum = root->val;
        getMaxSumPath(root, &max_sum);
        return max_sum;
    }
private:
    int getMaxSumPath(TreeNode* node, int* max_sum) {
        if (node == nullptr) return 0;
        auto l = max(getMaxSumPath(node->left, max_sum), 0);
        auto r = max(getMaxSumPath(node->right, max_sum), 0);
        *max_sum = max(*max_sum, node->val + l + r);
        return node->val + max(l, r);
    }
};
